src/gsCore/gsTemplateTools.h

```diff
@@ -233,13 +233,25 @@ typename make_signed<T>::type to_signed(T t) {
 template<class T1, class T2>
 bool less(T1 t1, T2 t2)
 {
-    if (is_signed<T1>::value == is_signed<T2>::value) // this is optimized at compile time
-        return (t1 < t2);
+    typedef typename util::make_signed<T1>::type signedT1;
+    typedef typename util::make_signed<T2>::type signedT2;
+    typedef typename util::make_unsigned<T1>::type unsignedT1;
+    typedef typename util::make_unsigned<T2>::type unsignedT2;
+
+    if (is_signed<T1>::value == is_signed<T2>::value) // all is_signed are optimized out at compile time
+    {
+        if (is_signed<T1>::value) // both signed, cast to signedTx
+            return (signedT1(t1) < signedT2(t2));
```

Comment:
Here, you are casting to T1 and T2, i.e., you are not casting.

Reply:
I'm not happy with the case:
```cpp
for (int i = 0; less(i, 10ull; ++i) {
    ...
} 
```
It looks like on godbolt that the compiler can't optimize it out - knows that i is positive - therefore goes trough the branches.

Btw. I read [Es.100 - 107](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es107-dont-use-unsigned-for-subscripts-prefer-gslindex) and now I do not know what would be best. Can we meet up?